protocol/src/main/java/org/geysermc/mcprotocollib/network/tcp/TcpPacketCodec.java

```diff
@@ -59,13 +72,19 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf buf, List<Object> out)
                 return;
             }
 
-            Packet packet = this.client ? packetProtocol.getPacketRegistry().createClientboundPacket(id, buf, codecHelper) : packetProtocol.getPacketRegistry().createServerboundPacket(id, buf, codecHelper);
+            log.debug("Decoding packet with id: {}", id);
+
+            Packet packet = this.client ? packetRegistry.createClientboundPacket(id, buf, codecHelper) : packetRegistry.createServerboundPacket(id, buf, codecHelper);
 
             if (buf.readableBytes() > 0) {
                 throw new IllegalStateException("Packet \"" + packet.getClass().getSimpleName() + "\" not fully read.");
             }
 
             out.add(packet);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Decoded packet: {}", packet.getClass().getSimpleName());
+            }
```

Comment:
True, but imo it should be kept like this in case of for example deadlocks. If the deserialization never returns a result or an error and instead infitely blocks the thread we'd have no idea what is going on.

Reply:
If it returns an error, that would be logged already, no? 

My concern would be flooding debug logs with double the lines for just a little more information that is, more or less, redundant

Reply:
If we really want this, can we set one to `fine`/`trace` logging level?

Reply:
Yeah ig trace works

Reply:
Done